scripts/run-clang-format.sh

```diff
@@ -25,8 +25,15 @@ SOURCE_EXT=(*.cc *.h)
 
 set -e
 
+# Detect run environment.
+if [ -n "$CI" ]; then
+  DRYRUN=" (dry run on CI)"
+elif [ -n "$GIT_AUTHOR_DATE" ]; then
+  DRYRUN=" (dry run in git hook)"
+fi
+
```

Comment:
I assume that 
- the `$CI` variable is (automatically) non-empty when the script is invoked from a CI/CD pipeline ?
- `$GIT_AUTHOR_DATE` is non-empty when the script is invoked from a pre-commit hook? 
- and neither of these two variables exist if the script is invoked as stand-alone?

Reply:
Exactly! I did not add any further comments, as the code already describes it. The $CI variable is documented at https://docs.gitlab.com/ee/ci/variables/predefined_variables.html . I found out $GIT_AUTHOR_DATE myself, by examining the environment when running a git commit hook.